code/datums/personal_statistics.dm

```diff
@@ -0,0 +1,565 @@
+				else
+					personal_statistics_user.self_heals++
+			is_healing = TRUE
+	return TRUE
+
+///Determine if a self or non-self heal occurred, and tally up the user mob's respective stat
+/obj/item/stack/medical/proc/record_healing(mob/living/user, mob/living/receiver)
+	if(!user.client)
+		return FALSE
+
+	var/datum/personal_statistics/personal_statistics_user = GLOB.personal_statistics_list[user.client]
+	if(user == receiver)
+		receiver = null
+
+	//If a receiving mob exists, we tally up to the user mob's stats that it performed a heal
+	if(receiver)
+		personal_statistics_user.heals++
+	else
+		personal_statistics_user.self_heals++
+	return TRUE
+
+///Record what was drank and if it was medicinal
+/obj/machinery/deployable/reagent_tank/proc/record_sippies(total, list/reagents_list, mob/user)
+	if(!total || !LAZYLEN(reagents_list) || !user.client)
+		return FALSE
+
+	var/is_healing
+	var/datum/personal_statistics/personal_statistics = GLOB.personal_statistics_list[user.client]
+	for(var/chem in reagents_list)
+		//Add the chem and amount consumed to the list
+		personal_statistics.chemicals_ingested[chem] += reagents_list[chem]
+		//Determine if a healing chem was involved; only needs to be done once
+		if(!is_healing && istype(chem, /datum/reagent/medicine))
+			personal_statistics.self_heals++
+			is_healing = TRUE
+	personal_statistics.sippies++
+	return TRUE
+
+///Tally up the corresponding weapon used by the pilot into their /datum/personal_statistics
+/obj/docking_port/mobile/marine_dropship/casplane/proc/record_cas_activity(obj/structure/dropship_equipment/weapon/weapon)
+	if(!chair.occupant.client)
+		return FALSE
+
+	var/datum/personal_statistics/personal_statistics = GLOB.personal_statistics_list[chair.occupant.client]
+	//Determine the type path of the weapon used and then tally accordingly
+	switch(weapon.type)
+		if(/obj/structure/dropship_equipment/weapon/heavygun)
+			personal_statistics.cas_cannon_shots++
+		if(/obj/structure/dropship_equipment/weapon/laser_beam_gun)
+			personal_statistics.cas_laser_shots++
+		if(/obj/structure/dropship_equipment/weapon/minirocket_pod)
+			personal_statistics.cas_minirockets_fired++
+		if(/obj/structure/dropship_equipment/weapon/rocket_pod)
+			personal_statistics.cas_rockets_fired++
+	return TRUE
+
+/***
+Refund the point costs of the parts in list/queue to a client's cas_points_used
+
+Note: this is probably the simplest solution with how the fabricator works.
+It may result in wonky stats if the person who ordered the parts was not the one to clear it,
+but rarely is a non-pilot the one to use it, let alone clear the queue.
+***/
+/obj/machinery/dropship_part_fabricator/proc/record_cas_point_refunds(mob/user)
+	if(!user.client)
+		return FALSE
+	var/datum/personal_statistics/personal_statistics = GLOB.personal_statistics_list[user.client]
+	for(var/item in queue)
+		//Why is the ammo not a child of this????
+		if(istype(item, /obj/structure/dropship_equipment))
```

Comment:
Why would ammo be lol

Reply:
You could have /weapon and /ammo children under /dropship_equipment. Just a little peeve since they both use the points system.